src/cache.rs

```diff
@@ -341,6 +341,31 @@ impl Cache {
                 }
             })
             .collect::<Vec<String>>();
+
+        if let Some(custom_pages_dir) = custom_pages_dir {
+            let is_page = |entry: &DirEntry| -> bool {
+                let path = entry.path();
+                let extension = &path.extension().and_then(OsStr::to_str).unwrap_or("");
+                entry.file_type().is_file() && extension == &"page"
+            };
+            let to_stem = |entry: DirEntry| -> Option<String> {
```

Comment:
We could move this above and use this function in the `pages` `filter_map` (line 334) too (we don't need the `let path = entry.path()` anymore then and can just use `entry.path()` in the line below)